logging/logrus/registry.go

```diff
@@ -55,22 +57,20 @@ func (lr *logRegistry) NewLogger(name string) logging.Logger {
 
 // ListLoggers returns a map (loggerName => log level)
 func (lr *logRegistry) ListLoggers() map[string]string {
-	lr.rwmutex.RLock()
-	defer lr.rwmutex.RUnlock()
+	mapping := lr.loggers.Load().(map[string]*Logger)
 	list := map[string]string{}
-	for k, v := range lr.mapping {
+	for k, v := range mapping {
 		list[k] = v.GetLevel().String()
 	}
 	return list
 }
 
 // SetLevel modifies log level of selected logger in the registry
 func (lr *logRegistry) SetLevel(logger, level string) error {
-	lr.rwmutex.RLock()
-	defer lr.rwmutex.RUnlock()
-	lg, ok := lr.mapping[logger]
+	mapping := lr.loggers.Load().(map[string]*Logger)
```

Comment:
let's have internal method that has mutex and returns a particular logger by name
then just call logger.SetLevel()